### ISSUE TITLE
output dimensions for vector quantize 

### ISSUE THREAD

**Issue by dorsa-zeinali** [lucidrains/vector-quantize-pytorch#168]
Hi, 
I have data that is the shape (batch, channel, height, width), I want to quantize it and I was wondering what the dimension of the vectorquantizer should be. I tried dim=height, but it produces something of dimensions [1, channel] when I need something of the dimension [batch, channel]. 
any help would be appreciated. 
thanks

**Comment by lucidrains** [comment]
could you show a short example script? I can make the correction for you easier that way

**Comment by dorsa-zeinali** [comment]
<img width="579" alt="Screenshot 2024-10-28 at 1 25 36 PM" src="https://github.com/user-attachments/assets/2556b5ce-a725-43f9-ad3f-899c45002a73">
Here is the description of my model. my code is rather long , should I include other information? although I solved the issue by setting the batch size to 1. If I train this network, will it also update the codebook with the optimizer? or will the updates just be ema ?
